src/app/shared/components/play-controls/play-controls.component.ts

```diff
@@ -1,57 +1,59 @@
-import {  Component,  ElementRef,  ViewChild,} from "@angular/core";
-import { PlaybackService } from "../../../core/models/services/play/playback.service";
+import { PlaybackService } from './../../../core/models/services/play/playback.service';
+import { Component, ElementRef, ViewChild } from "@angular/core";
 import { Store } from "@ngrx/store";
-import {  nextTrack,  pauseTrack,  playTrack,  previousTrack,  trackEnded,} from "../../../store/play-track/playMusic.actions";
-import { selectCurrentTrack } from "../../../store/play-track/playMusic.selectors";
+
+import { SpotifyPlayerService } from "../../../core/models/services/spotify/spotify-player.service";
+import { PlayingSongState } from "../../../core/models/interfaces/spotify";
 
 @Component({
   selector: "app-play-controls",
   templateUrl: "./play-controls.component.html",
   styleUrl: "./play-controls.component.css",
 })
 export class PlayControlsComponent {
-  @ViewChild("audioPlayer") audioPlayerRef!: ElementRef<HTMLAudioElement>;
-  trackDetails = this.store.selectSignal(selectCurrentTrack);
-
-  audio = new Audio();
-  isPlaying: any;
-  constructor(private playbackService: PlaybackService, private store: Store) {}
-
-  ngOnInit(): void {
-    console.log(this.trackDetails());
-  }
 
+  @ViewChild("audioPlayerRef") audioPlayerRef!: ElementRef<HTMLAudioElement>;
+  isPlaying = false;
+  trackDetail!: PlayingSongState | null;
```

Comment:
remove the null since you're telling typescript that trackDetail cannot be null.